iOS/AppleSiliconSimulator.py

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env python3
+import fileinput
+import os
+import subprocess
+import sys
+import xml.etree.ElementTree as ET
+
+'''
+Update an iOS XCFramework that supports device and X86_64 simulator to instead support device an
+ARM64 simulator. This is designed to work with IModelJsNative.xcframework.
+'''
+
+def get_versions(src_path: str) -> tuple[str]:
+    '''
+    Return the minios and SDK versions of the given framework binary file. Note that if the binary
+    does not include an explicit minios version, the SDK version will be used.
+    '''
+    result = subprocess.run(['xcrun', 'vtool', '-arch', 'arm64', '-show', src_path], capture_output=True, text=True)
+    minos = None
+    sdk = None
+    minos_prefix = 'minos '
+    sdk_prefix = 'sdk '
+    for line in result.stdout.splitlines():
+        stripped = line.strip()
+        if stripped.startswith(minos_prefix):
+            minos = stripped[len(minos_prefix):]
+        if stripped.startswith(sdk_prefix):
+            sdk = stripped[len(sdk_prefix):]
+    if sdk is not None:
+        if minos is not None:
+            return minos, sdk
+        else:
+            return sdk, sdk
+    raise Exception("Cannot find SDK version in XCFramework!")
+
+def run_command(command: str, error: str | None = None) -> None:
+    '''
+    Call `os.system` with the specified command and throw an exception with the given error string
+    if the exit code from the command is non-zero.
+    '''
+    if os.system(command) != 0:
+        raise Exception(error or f'Error executing command {command}!')
+
+def copy_directory_tree(src: str, dst: str) -> None:
+    '''
+    Copies a directory tree from one place to another.
+    '''
+    run_command(f'rsync -a \'{os.path.dirname(src)}/\' \'{os.path.dirname(dst)}/\'', 'Error copying arm64 framework!')
```

Comment:
You could consider using [copytree](https://docs.python.org/3/library/shutil.html#copytree-example) instead?

Reply:
Fixed.